inc/features/blocks/experience/register.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * Experience Block Server Side.
+ *
+ * @phpcs:ignoreFile HM.Files.FunctionFileName.WrongFile
+ */
+
+namespace Altis\Experiments\Features\Blocks\Experience;
+
+use const Altis\Experiments\ROOT_DIR;
+use function Altis\Experiments\Features\Blocks\get_block_settings;
+
+const BLOCK = 'experience';
+
+function setup() {
+	$block_data = get_block_settings( BLOCK );
+
+	// Queue up JS files.
+	add_action( 'enqueue_block_editor_assets', __NAMESPACE__ . '\\enqueue_assets' );
+
+	// Register the block.
+	register_block_type( $block_data['name'], [
+		'attributes' => $block_data['settings']['attributes'],
+		'render_callback' => __NAMESPACE__ . '\\render_block',
+	] );
+}
+
+function enqueue_assets() {
+	wp_enqueue_script(
+		'altis-experiments-features-blocks-experience',
+		plugins_url( 'build/features/blocks/experience.js', ROOT_DIR . '/plugin.php' ),
+		[
+			'wp-plugins',
+			'wp-blocks',
+			'wp-i18n',
+			'wp-editor',
+			'wp-components',
+			'wp-core-data',
+			'wp-edit-post',
+			'altis-experiments-features-blocks-variant',
+			// Exports the audience UI modal component.
+			'altis-analytics-audience-ui',
+		]
+	);
+
+	// Queue up editor CSS.
+	wp_enqueue_style(
+		'altis-experiments-features-blocks-experience',
+		plugins_url( 'inc/features/blocks/experience/edit.css', ROOT_DIR . '/plugin.php' )
+	);
+}
+
+/**
+ * Render callback for the experience block.
+ *
+ * Because this block only saves <InnerBlocks.Content> on the JS side,
+ * the content string represents only the wrapped inner block markup.
+ *
+ * @param array  $attributes   The block's attributes object.
+ * @param string $innerContent The block's saved content.
+ * @return string The final rendered block markup, as an HTML string.
+ */
+function render_block( array $attributes, ?string $inner_content = '' ) : string {
+	$client_id = $attributes['clientId'] ?? false;
```

Comment:
What is the `clientId`? I don't really see anywhere that this concept is documented.

Reply:
In the absence of a post ID the `clientId` is generated by the block editor - every block gets a uuid when being edited. We're taking the initial one and storing it as an attribute so that the experience block component and the variant templates can be associated easily.

Reply:
Ahh ok cool - so clientId is a gberg concept?

Reply:
Yep, correct. I think we could consider implicitly converting these into reusable blocks by default though as it'll make the AB testing side of it much much easier. Also easier to show where they're used and the analytics data associated with them rather than having to find the block and the post and selecting it.